pages/api/friends.js

```diff
@@ -1,27 +1,67 @@
 const { Friend, User } = require('../../models');
+const Promise = require('bluebird');
+const _ = require('lodash');
 
 export default async (req, res) => {
   if (req.method === 'POST') {
     let user_id = req.body.user_id;
     let user = await User.findByPk(user_id);
+    let names;
 
-    if (user) {
-      let existingFriend = await Friend.findAll({
-        where: { user_id: user.id, name: req.body.name },
+    if (!req.body.name) {
+      res.status(400).json({
+        httpCode: 400,
+        message: 'Must provide name(s)',
       });
+    } else {
+      names =
+        typeof req.body.name === 'string' ? [req.body.name] : req.body.name;
 
-      if (
-        existingFriend &&
-        existingFriend.length &&
-        existingFriend[0].name === req.body.name
-      ) {
-        return res.status(409).json({
-          httpCode: 409,
-          message: 'Friend with this name already exists',
+      if (names.length !== _.uniq(names).length) {
+        res.status(400).json({
+          httpCode: 400,
+          message: `Please remove duplicate friend names`,
         });
-      } else {
-        res.json(await Friend.create(req.body));
       }
+    }
+
+    if (user) {
+      let successful = [];
+      let errored = [];
+
+      Promise.mapSeries(names, async (name) => {
```

Comment:
What do you think of clearing the existing friends and adding the ones in `req.body.name`?